ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -199,7 +201,8 @@ data Handlers m addr blk = Handlers {
 
 mkHandlers
   :: forall m blk addrNTN addrNTC.
-     ( IOLike m
+     ( HasHardForkHistory blk
```

Comment:
Every block should be able to instantiate this class, so I don't think this addition is a detriment.